src/GafferScene/MotionPath.cpp

```diff
@@ -0,0 +1,380 @@
+{
+	if( !adjustBoundsPlug()->getValue() )
+	{
+		h = inPlug()->boundPlug()->hash();
+		return;
+	}
+
+	FilteredSceneProcessor::hashBound( path, context, parent, h );
+
+	filterHash( context, h );
+
+	outPlug()->objectPlug()->hash( h );
+	outPlug()->childBoundsPlug()->hash( h );
+}
+
+Imath::Box3f MotionPath::computeBound( const ScenePath &path, const Context *context, const ScenePlug *parent ) const
+{
+	if( !adjustBoundsPlug()->getValue() )
+	{
+		return inPlug()->boundPlug()->getValue();
+	}
+
+	const PathMatcher::Result m = filterValue( context );
+	if( m & ( PathMatcher::ExactMatch | PathMatcher::DescendantMatch ) )
+	{
+		Imath::Box3f result = outPlug()->childBoundsPlug()->getValue();
+		if( m & PathMatcher::ExactMatch )
+		{
+			if( ConstCurvesPrimitivePtr motionPath = runTimeCast<const CurvesPrimitive>( outPlug()->objectPlug()->getValue() ) )
+			{
+				result.extendBy( SceneAlgo::bound( motionPath.get() ) );
+			}
+		}
+		return result;
+	}
+
+	return outPlug()->childBoundsPlug()->getValue();
+}
+
+void MotionPath::hashTransform( const ScenePath &path, const Context *context, const ScenePlug *parent, MurmurHash &h ) const
+{
+	h = MurmurHash();
+	h.append( inPlug()->transformPlug()->defaultValue() );
+}
+
+Imath::M44f MotionPath::computeTransform( const ScenePath &path, const Context *context, const ScenePlug *parent ) const
+{
+	return inPlug()->transformPlug()->defaultValue();
+}
+
+void MotionPath::hashAttributes( const ScenePath &path, const Context *context, const ScenePlug *parent, MurmurHash &h ) const
+{
+	h = inPlug()->attributesPlug()->defaultValue()->Object::hash();
+}
+
+ConstCompoundObjectPtr MotionPath::computeAttributes( const ScenePath &path, const Context *context, const ScenePlug *parent ) const
+{
+	return inPlug()->attributesPlug()->defaultValue();
+}
+
+void MotionPath::hashObject( const ScenePath &path, const Context *context, const ScenePlug *parent, MurmurHash &h ) const
+{
+	FilteredSceneProcessor::hashObject( path, context, parent, h );
+
+	filterHash( context, h );
+
+	h.append( inPlug()->fullTransformHash( path ) );
+
+	h.append( context->getFrame() );
+
+	startModePlug()->hash( h );
+	startFramePlug()->hash( h );
+	endModePlug()->hash( h );
+	endFramePlug()->hash( h );
+	incrementPlug()->hash( h );
+}
+
+ConstObjectPtr MotionPath::computeObject( const ScenePath &path, const Context *context, const ScenePlug *parent ) const
+{
+	if( path.empty() || !(filterValue( context ) & PathMatcher::ExactMatch) )
+	{
+		return inPlug()->objectPlug()->defaultValue();
+	}
+
+	float start = ( (FrameMode)startModePlug()->getValue() == FrameMode::Absolute ) ? startFramePlug()->getValue() : context->getFrame() + startFramePlug()->getValue();
+	float end = ( (FrameMode)endModePlug()->getValue() == FrameMode::Absolute ) ? endFramePlug()->getValue() : context->getFrame() + endFramePlug()->getValue();
+	if( start >= end )
+	{
+		return inPlug()->objectPlug()->defaultValue();
+	}
```

Comment:
I'm usually the one in favour of errors, but I don't actually like an error much here, because of the "motion graphics" case where someone is just animating the start and end to look cool.  I don't think I actually mind the current behaviour much, it seems like someone who turns it inside out should probably be capable of figuring out what they've done, but I guess flipping increment is OK too.

Reply:
I thought the standard was no-errors on filtered nodes, I'd originally had this throw and then back-peddled because I thought I was supposed to. Could be a Warning though I suppose.

Reply:
The "no errors" guideline is motivated by the "swiss army knife" workflows people want, where you're trying to build a graph that will work with any input scene/shot/bundle/whatever. I didn't think that applied here really - it mostly applies to the existence of locations/primvars. But I don't feel strongly about this. If we don't have a consensus, let's leave as-is and be guided by anything that comes up in the real world.